src/test/interface/io/json/json_parser_test.cpp

```diff
@@ -363,231 +363,187 @@ TEST(ioJson,jsonParserErr01) {
 }
 
 
 TEST(ioJson,jsonParserErr12d) {
   test_exception("{ \"a\" : 5 ] }",
-                 "expecting end of object } or separator ,\n");
+                 "Missing a comma or '}' after an object member.\n");
 }
 
 TEST(ioJson,jsonParserErr12e) {
   test_exception("{ \"a\" : [ 5 ] ] }",
-                 "expecting end of object } or separator ,\n");
+                 "Missing a comma or '}' after an object member.\n");
 }
 
 TEST(ioJson,jsonParserErr13) {
   test_exception("{ hello }",
-                 "expecting member key or end of object marker (})\n");
+                 "Missing a name for object member.\n");
 }
 
 
 TEST(ioJson,jsonParserErr14) {
   test_exception("{ \"foo\": -1.0100e09 , }",
-              "expecting member key or end of object marker (})\n");
+              "Missing a name for object member.\n");
 }
 
 
 TEST(ioJson,jsonParserErr14a) {
   test_exception("{ { \"foo\": -1.0100e09 , }",
-              "expecting member key or end of object marker (})\n");
+              "Missing a name for object member.\n");
 }
 
 
 TEST(ioJson,jsonParserErr14b) {
   test_exception("{ \"bar\" : { \"foo\": -1.0100e09 , }",
-              "expecting member key or end of object marker (})\n");
+              "Missing a name for object member.\n");
 }
 
 
 TEST(ioJson,jsonParserErr14c) {
   test_exception("{ \"bar\" : [ \"foo\": -1.0100e09 , }",
-                 "in array, expecting ] or ,\n");
+                 "Missing a comma or ']' after an array element.\n");
 }
 
 TEST(ioJson,jsonParseErr14d) {
   test_exception("{  \"foo\" : [ { \"bar\": { \"baz\": [ 1, 2]  } }, -3, -4.44  }  ",
-                 "in array, expecting ] or ,\n");
+                 "Missing a comma or ']' after an array element.\n");
 }
 
 TEST(ioJson,jsonParseErr14e) {
   test_exception("{  \"foo\" : [ { \"bar\": { \"baz\": [ 1, 2]  } }, -3, -4.44 } } } ] }  ",
-                 "in array, expecting ] or ,\n");
+                 "Missing a comma or ']' after an array element.\n");
 }
 
 TEST(ioJson,jsonParserErr14f) {
   test_exception("{ \"foo\": -1.0100e09 , ",
-              "unexpected end of stream\n");
+              "Missing a name for object member.\n");
 }
 
 
 TEST(ioJson,jsonParserErr15) {
   test_exception("{ \"5\" 5 }",
-                 "expecting key-value separator :\n");
+                 "Missing a colon after a name of object member.\n");
 }
 
 TEST(ioJson,jsonParserErr16) {
   test_exception("{ \"5\" :  5  \"6\" : 6 }",
-                 "expecting end of object } or separator ,\n");
+                 "Missing a comma or '}' after an object member.\n");
 }
 
 TEST(ioJson,jsonParserErr17) {
   test_exception("{ \"5\" : ",
-                 "unexpected end of stream\n");
+                 "Invalid value.\n");
 }
 
 TEST(ioJson,jsonParserErr18) {
   test_exception("[ -1, -2, \"-inf\", ]",
-                 "in array, expecting value\n");
-}
-
-
-TEST(ioJson,jsonParserErr19a) {
-  test_exception("[ 1111111111111111111111111111 ]",
```

Comment:
This gets parsed as a double and it doesnt throw.

Reply:
is that number within double range?  can you craft a number that's bigger than max double value?

Reply:
Yes, 310 ones do it. That throws "Number too big to be stored in double.\n"